src/GafferScene/ShaderPlug.cpp

```diff
@@ -139,12 +119,43 @@ bool ShaderPlug::acceptsInput( const Gaffer::Plug *input ) const
 
 IECore::MurmurHash ShaderPlug::attributesHash() const
 {
-	const Shader *s = shader( this );
-	return s ? s->attributesHash() : MurmurHash();
+	const Gaffer::Plug *p = shaderOutPlug();
+
+	IECore::MurmurHash h;
+	if (p)
+	{
+		const GafferScene::Shader *s = runTimeCast<const GafferScene::Shader>( p->node() );
```

Comment:
I don't quite follow this comment so I'm not sure what check to do `ShaderPlug::shaderOutPlug` as I'm casting to a `GafferScene::Shader` rather than `ShaderPlug` ?